src/Presentation/TabsHtmlBuilder.php

```diff
@@ -52,12 +55,13 @@ private function parseQuery( string $searchQuery ): Query {
 	}
 
 	private function buildSettingsViewModel(): array {
+		if ( !$this->enableWikiConfig ) {
+			return [];
+		}
+
 		$title = Title::newFromText( WikibaseFacetedSearchExtension::CONFIG_PAGE_TITLE, NS_MEDIAWIKI );
 
-		if (
-			!$title instanceof Title
-			|| !$this->user->isAllowed( 'editsitejson' )
-		) {
+		if ( !$title instanceof Title || !$this->permissionManager->userCan( 'edit', $this->user, $title ) ) {
```

Comment:
The above nicely fits into a `ConfigurationAuthorizer` service which is injected here and itself takes 
* bool $enableWikiConfig,
* PermissionManager $permissionManager,
* User $user
* TitleFactory or whatever it is called, to avoid `Title::newFromText`

Adding multiple new dependencies to an existing service for one reason is a sign that _suggests_ a new service is called for

Reply:
Plus, having 7 fields is also slightly suspect. It can happen, but most of the time, classes in a well designed codebase will have less. 7 fields is similar to 3 or 4 function parameters